main/lsp/ExtractVariable.cc

```diff
@@ -90,6 +90,18 @@ class LocSearchWalk {
             return;
         }
 
+        // If the new enclosingScope is an If node, and else branch is another if node, then this is probably an elsif
```

Comment:
> probably

Do you have a test that shows that this heuristic breaks if the user actually wrote

```ruby
if cond
else
  if something_to_extra
  end
end
```

?

Reply:
Added a new test case for that.